File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    days = []
    
    for x,y in zip(progresses, speeds):
        day = (100-x) // y
        if (100-x) % y != 0:
            day += 1
        days.append(day)
    
    current_day = days[0]
    count = 0
    for day in days:
        if day <= current_day: 
            count +=1
        else:
            answer.append(count)
            current_day = day
            count = 1
        
    answer.append(count)
    
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    var answer = [];
    var days = [];
    
    for (let i = 0; i<progresses.length ; i++){
        var day = Math.floor((100 - progresses[i]) / speeds[i]);

        if ((100 - progresses[i]) % speeds[i] !== 0) {
            day += 1;
        }
        days.push(day)
    }
    
    var current_day = days[0]
    count = 0
    
    for (let i = 0; i<days.length; i++){
        if (days[i] <= current_day){
            count += 1
        } else{
            answer.push(count)
            current_day = days[i]
            count = 1
        }
    }
    answer.push(count)
    
    return answer;
}